### PR TITLE
[SNOW-1297195] Add tox to daily jenkins job, fix tox.ini.

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-1297195

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   There were a few issues which prevent the daily snowpark pandas API jenkins job to run successfully:
      - tox.ini had `MODIN_PYTEST_NO_COV_CMD` missing
      - the jenkins invocation script requires tox to be installed
      - the tox_install_cmd.sh was using sh, but should be bash and variables were not defined.
   this PR addresses these issues.

### CODE REVIEW

**Review by sfc-gh-nkrishna — APPROVED** [review]

**Review by noloerino — APPROVED** [review]

**Review by sfc-gh-joshi — APPROVED** [review]

File: tox.ini [hunk 1]
```diff
@@ -62,6 +62,7 @@ setenv =
     SNOW_1314507_WORKAROUND_RERUN_FLAGS = --reruns 2 --only-rerun "Insufficient resource during interleaved execution."
     MODIN_PYTEST_CMD = pytest {env:SNOWFLAKE_PYTEST_VERBOSITY:} {env:SNOWFLAKE_PYTEST_PARALLELISM:} {env:SNOWFLAKE_PYTEST_COV_CMD} --ignore=tests/resources
     MODIN_PYTEST_DAILY_CMD = pytest {env:SNOWFLAKE_PYTEST_VERBOSITY:} {env:SNOWFLAKE_PYTEST_DAILY_PARALLELISM:} {env:SNOWFLAKE_PYTEST_COV_CMD} --ignore=tests/resources
+    MODIN_PYTEST_NO_COV_CMD = pytest {env:SNOWFLAKE_PYTEST_VERBOSITY:} {env:SNOWFLAKE_PYTEST_PARALLELISM:} --ignore=tests/resources
```
Comment (sfc-gh-azhan) [inline tox.ini:65]: where is it used? please add comment

File: scripts/tox_install_cmd.sh [hunk 2]
```diff
@@ -13,6 +13,9 @@ done
 
 echo "${pip_options[*]}"
 
+# Default to empty, to ensure snowflake_path variable is defined.
+snowflake_path=${snowflake_path:-""}
```
Comment (sfc-gh-azhan) [inline scripts/tox_install_cmd.sh:17]: why we need this?

File: scripts/tox_install_cmd.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env sh
+#!/usr/bin/env bash
```
Comment (sfc-gh-azhan) [inline scripts/tox_install_cmd.sh:1]: why need bash?

File: tox.ini (hunk 1 above)
Comment (sfc-gh-nkrishna) [reply tox.ini:65]: There is a comment where it is used: https://github.com/snowflakedb/snowpark-python/pull/1442/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R96-R98.

File: scripts/tox_install_cmd.sh (hunk 2 above)
Comment (sfc-gh-lspiegelberg) [reply scripts/tox_install_cmd.sh:17]: I got an error exit code 2, unbound variables. This fixes it.

File: scripts/tox_install_cmd.sh (hunk 3 above)
Comment (sfc-gh-lspiegelberg) [reply scripts/tox_install_cmd.sh:1]: we actually call the script via `bash`. `bash` has different syntax than `sh`, correcting the line to reflect the shell the script is written for.

**Pull request merged**